File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        return logLine.Split(['[',']',':'])[3].Trim();
    }

    public static string LogLevel(string logLine)
    {
        return logLine.Split(['[',']',':'])[1].ToLower();
    }

    public static string Reformat(string logLine)
    {
        string logType;
        string logMessage;

        char[] delimiters = ['[',']',':'];
        string[] substrings = logLine.Split(delimiters);

        logType = substrings[1].ToLower();
        logMessage = substrings[3].Trim();

        return logMessage + " (" + logType + ")";
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
	int start = line.find(":") + 2;
	int size = line.length() - start;

	return line.substr(start,size);
}

std::string log_level(std::string line) {
	int open = line.find("[") +1;
	int close = line.find("]");
	int lenght = close-open;

	return line.substr(open,lenght);
}

std::string reformat(std::string line) {
	return message(line) + " (" + log_level(line) + ")";
}
}  // namespace log_line
